src/tests/Microsoft.PowerFx.Core.Tests/DisplayNameTests.cs

```diff
@@ -18,6 +20,80 @@ namespace Microsoft.PowerFx.Interpreter.Tests
 {
     public class DisplayNameTests : PowerFxTest
     {
+        public class LazyRecordType : RecordType
+        {
+            public override IEnumerable<string> FieldNames { get; }
+
+            public override bool TryGetFieldType(string name, out FormulaType type)
+            {
+                type = name switch
+                {
+                    "Num" => FormulaType.Number,
+                    "B" => FormulaType.Boolean,
+                    "Nested" => FormulaType.Unknown,
```

Comment:
If this is intended to mimic the pre-defined record, this doesn't quite match. The Nested field is a TableType that contains `Inner`. Inner is not a member of the top level type.

Reply:
Jas - can you add a test that illustrates this difference?

Reply:
Calling Check against both r1/r2 would show this.